ingestors/nuget.go

```diff
@@ -0,0 +1,132 @@
+package ingestors
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/librariesio/depper/data"
+)
+
+const nugetSchedule = "18 * * * *"
+const nugetIndexUrl = "https://api.nuget.org/v3/catalog0/index.json"
+
+type nugetIndex struct {
+	IndexId string `json:"@id"`
+	Pages   []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+	} `json:"items"`
+}
+
+type nugetPage struct {
+	PageId   string `json:"@id"`
+	Packages []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+		Name            string `json:"nuget:id"`
+		Version         string `json:"nuget:version"`
+	} `json:"items"`
+}
+
+type Nuget struct {
+	LatestRun time.Time
+}
+
+func NewNuget() *Nuget {
+	return &Nuget{}
+}
+
+func (ingestor *Nuget) Schedule() string {
+	return nugetSchedule
+}
+
+func (ingestor *Nuget) Ingest() []data.PackageVersion {
+	// Until we save LatestRun state, begin with the last 24 hours.
```

Comment:
My general gut is you want to go back more than one window of time in case there was a missed run (and to allow for off-by-some-splay). So if we shift the syncing to be every hour (for example), we probably want to look at 2.5-3 hours of data.

Reply:
Good point, only really needs to account for restarts (or maybe outages).

Reply:
(just remembering now that I did a big window because I thought that the packages for each page were set to the same "updated at" timestamp, so they were all lumped into big time chunks, but now I realize I was reading the wrong timestamp!)